wp-env: Add --quiet option to destroy command

jrmd opened the pull request:
## What?
This patch adds support for `wp-env destroy --quiet` which allows for the command to be ran silently and without the blocking confirmation prompt.

## Why?

I am currently creating a script to run e2e tests across numerous plugins/repos all with their own individual e2e tests set up that need to be ran independently, when trying to run `wp-env destroy` within the script it requires a confirmation from stdin, forcing me to to write to stdin in a rather clunky way. By adding the skip confirmation option allows us to bypass this limitation allows the script to cleanup without issue.

## How?
PR adds an optional argument to the destroy command which is used to silence all non-error output and skip blocking confirmation prompts. This patch lays some of the necessary foundation to implement across other commands too.

## Testing Instructions
1. run `wp-env destroy --quiet`
2. see the confirmation isn't shown and the environment is destroyed as expected
3. run `wp-env destroy`
4. see the confirmation prompt is shown and works as previous

jrmd committed:
feat(wp-env): add support for --skip-confirmation option

This patch adds support for `wp-env destroy --skip-confirmation` which
allows for the command to be ran without the blocking confirmation
prompt. Which easily allows this to be ran within scripts and on CI.

jrmd committed:
style: fix linting issues

jrmd committed:
Merge branch 'WordPress:trunk' into feature/wp-env-skip-confirmation

ObliviousHarmony commented:
I like this @jrmd, thank you! Before I test the PR out though, how would you feel about sticking to a more standard option like `--silent/-s` or `--quiet/-q`? I would probably lean towards silencing non-error output and auto-accepting any confirmation for the related command (if there is any).

jrmd commented:
@ObliviousHarmony definitely seems sensible to me :) i will make the necessary changes and update the pr :)

jrmd committed:
feat: use -q,--quiet instead of skip-confirmation

This patch changes the skip-confirmation option for wp-env destroy to
`--quiet` which also silences any non-error output along with auto
confirming options.

This change required a change in the withSpinner HOC due to the
.succeed at the end outputting the confirmation message but it does help
us in future to implement something for other commands

jrmd commented:
@ObliviousHarmony Updated the pr to use the --quiet/-q 😄 let me know if you have any questions or issues.

jrmd committed:
test(wp-env): update the ora mock to make it return in an instance

ora itself returns an instance, with the addition of `--quiet` spinner
is no longer assigned to `spinner.start()` this mock has been updated to
return an instance to itself.
